### ISSUE TITLE
Adds additional column into generated query on include

### ISSUE THREAD

**Issue by davemanton** [dotnet/efcore#22853]
I am finding that when I create a query to retrieve data from two tables in ef core, the generated query contains an additional attribute which does not exist on the table it is querying.  The same query and equivalent mapping works in Entity Framework 6.3

I am trying to retrieve the contents of an attribute table and include the contents of the attribute relationships table which it has two foreign key relationships to.  See the screenshot below

![image](https://user-images.githubusercontent.com/10618401/94692833-ed51c180-032a-11eb-90ec-3575659f4885.png)

The following is a basic example of the equivalent query I'm using, I've tried it with this exact code and it generates the exact same issue as well. The code is very simple as this is still under development and while I do put a where clause on as well it doesn't work with or without the where clause

```C#
var exampleQuery = _myContext.Attributes.Include(x => x.AncestorOfRelationships).ToList()
```

This generates the following sql
```SQL
SELECT [a].[AttributeId]
	 , [a].[AttributeTypeId]
	 , [a].[PropertyName]
	 , [a0].[AttributeRelationshipId]
	 , [a0].[AncestorAttributeId]
	 , [a0].[AttributeId]
	 , [a0].[AttributeRelationshipTypeId]
	 , [a0].[DescendantAttributeId]
	 , [a0].[ParentAttributeRelationshipId]
 FROM [dbo].[Attributes] AS [a]
 LEFT JOIN [dbo].[AttributeRelationships] AS [a0] 
	ON [a].[AttributeId] = [a0].[AncestorAttributeId]
 ORDER BY [a].[AttributeId], [a0].[AttributeRelationshipId]
```

The issue is that [a0].[AttributeId] doesn't exist on the [dbo].[AttributeRelationships] AS [a0] table

Mapping Code for entities
```C#
public class AttributeMap : IEntityTypeConfiguration<Attribute>
    {
        public void Configure(EntityTypeBuilder<Attribute> builder)
        {
            builder.ToTable("Attributes", "dbo");
            builder.HasKey(x => x.AttributeId).HasName("PK_Attributes");

            builder.Property(x => x.AttributeId).HasColumnName(@"AttributeId").HasColumnType("int").IsRequired().ValueGeneratedOnAdd().UseIdentityColumn();
            builder.Property(x => x.PropertyName).HasColumnName(@"PropertyName").HasColumnType("varchar(100)").IsRequired().IsUnicode(false).HasMaxLength(100);

            builder.Ignore(x => x.StructuralDescendants);

        }
    }

public class AttributeRelationshipMap : IEntityTypeConfiguration<AttributeRelationship>
    {
        public void Configure(EntityTypeBuilder<AttributeRelationship> builder)
        {
            builder.ToTable("AttributeRelationships", "dbo");
            builder.HasKey(x => x.AttributeRelationshipId).HasName("PK_AttributeRelationships");

            builder.Property(x => x.AttributeRelationshipId).HasColumnName(@"AttributeRelationshipId").HasColumnType("int").IsRequired().ValueGeneratedOnAdd().UseIdentityColumn();
            builder.Property(x => x.ParentAttributeRelationshipId).HasColumnName(@"ParentAttributeRelationshipId").HasColumnType("int").IsRequired(false);
            builder.Property(x => x.AncestorAttributeId).HasColumnName(@"AncestorAttributeId").HasColumnType("int").IsRequired();
            builder.Property(x => x.DescendantAttributeId).HasColumnName(@"DescendantAttributeId").HasColumnType("int").IsRequired();
            builder.Property(x => x.AttributeRelationshipTypeId).HasColumnName(@"AttributeRelationshipTypeId").HasColumnType("int").IsRequired();

            // Foreign keys
            builder.HasOne(a => a.AncestorAttribute).WithMany(b => b.AncestorOfRelationships).HasForeignKey(c => c.AncestorAttributeId).OnDelete(DeleteBehavior.ClientSetNull).HasConstraintName("FK_AncestorAttributeRelationships_Attributes");
            builder.HasOne(a => a.DescendantAttribute).WithMany(b => b.DescendantOfRelationships).HasForeignKey(c => c.DescendantAttributeId).OnDelete(DeleteBehavior.ClientSetNull).HasConstraintName("FK_DescendantAttributeRelationships_Attributes");
            builder.HasOne(a => a.AttributeRelationshipType).WithMany(b => b.AttributeRelationships).OnDelete(DeleteBehavior.ClientSetNull).HasConstraintName("FK_AttributeRelationships_AttributeRelationshipTypes");
            builder.HasOne(a => a.ParentAttributeRelationship).WithMany(b => b.ChildAttributeRelationships).HasForeignKey(c => c.ParentAttributeRelationshipId).OnDelete(DeleteBehavior.ClientSetNull).HasConstraintName("FK_ParentAttributeRelationships_AttributeRelationships");

            builder.HasIndex(x => new { x.ParentAttributeRelationshipId, x.AncestorAttributeId, x.DescendantAttributeId }).HasDatabaseName("UQ_AttributeRelationships").IsUnique();
        }
    }
```

Code for Domain - a key point here is that attribute uses the attributeTypeId as a discriminator which isn't included on the domain objects

```C#
public abstract class Attribute
    {
        protected Attribute()
        {
            AttributeOptions = new HashSet<AttributeOption>();
            AncestorOfRelationships = new HashSet<AttributeRelationship>();
            DescendantOfRelationships = new HashSet<AttributeRelationship>();
            AttributeDisplayLabels = new HashSet<AttributeDisplayLabel>();
        }

        protected Attribute(string propertyName,
                            IDictionary<CultureInfo, string> displayNames,
                            IDictionary<Attribute, AttributeRelationshipType>? relationships,
                            IEnumerable<AttributeOptionValue>? options) : this()
        {
            PropertyName = propertyName;

            foreach(var name in displayNames)
                AttributeDisplayLabels.Add(new AttributeDisplayLabel(this, name.Key, name.Value));

            if (relationships != null)
                foreach (var relationship in relationships)
                {
                    var attributeRelationship = new AttributeRelationship(this, relationship.Key, relationship.Value);
                    AncestorOfRelationships.Add(attributeRelationship);
                }

            if (options == null)
                return;

            foreach(var option in options)
                AttributeOptions.Add(new AttributeOption(this, option.OptionIndex, option.CultureInfo, option.Value));
        }

        public int AttributeId { get; set; } = default!;
        public string PropertyName { get; set; } = default!;

        public ICollection<AttributeOption> AttributeOptions { get; set; }
        public ICollection<AttributeRelationship> AncestorOfRelationships { get; set; }
        public ICollection<AttributeRelationship> DescendantOfRelationships { get; set; }
        public ICollection<AttributeDisplayLabel> AttributeDisplayLabels { get; set; }
    }

public class StringAttribute : Attribute
    {
        private StringAttribute() { }

        internal StringAttribute(string propertyName,
                                 IDictionary<CultureInfo, string> displayNames,
                                 IDictionary<Attribute, AttributeRelationshipType>? relationships,
                                 IEnumerable<AttributeOptionValue>? options)
            : base(propertyName, displayNames, relationships, options) { }
    }

public class AttributeRelationship
    {
        private AttributeRelationship()
        {
            ChildAttributeRelationships = new HashSet<AttributeRelationship>();
        }

        private AttributeRelationship(AttributeRelationship parentRelationship,
                                      Attribute ancestor,
                                      Attribute descendant,
                                      AttributeRelationshipType relationshipType)
            : this(ancestor, descendant, relationshipType)
        {
            ParentAttributeRelationshipId = parentRelationship.AttributeRelationshipId;
            ParentAttributeRelationship = parentRelationship;
        }

        internal AttributeRelationship(Attribute ancestor,
                                       Attribute descendant,
                                       AttributeRelationshipType relationshipType) 
            : this()
        {
            AncestorAttribute = ancestor;
            AncestorAttributeId = ancestor.AttributeId;

            DescendantAttribute = descendant;
            DescendantAttributeId = descendant.AttributeId;

            AttributeRelationshipTypeId = relationshipType.AttributeRelationshipTypeId;

            // Commented out as an unnecessary property for example is used here to create relationships
            //foreach (var descendantRelationship in descendant.StructuralDescendants)
            //{
            //    var relationship = new AttributeRelationship(this, ancestor, descendantRelationship.DescendantAttribute, AttributeRelationshipType.__INTERNAL_ONLY_ANCESTOR_OF);
            //    ChildAttributeRelationships.Add(relationship);
            //}
        }

        public int AttributeRelationshipId { get; private set; }
        public int? ParentAttributeRelationshipId { get; private set; }
        public int AncestorAttributeId { get; private set; }
        public int DescendantAttributeId { get; private set; }
        public int AttributeRelationshipTypeId { get; private set; }

        public AttributeRelationship? ParentAttributeRelationship { get; private set; }
        public Attribute AncestorAttribute { get; private set; } = default!;
        public Attribute DescendantAttribute { get; private set; } = default!;
        public AttributeRelationshipType? AttributeRelationshipType { get; private set; }

        public ICollection<AttributeRelationship> ChildAttributeRelationships { get; private set; }
    }
```

Relevant parts of the Context

```C#
public class ProductInformationContext : DbContext
    {
        public ProductInformationContext()
        {
        }

        public ProductInformationContext(DbContextOptions<ProductInformationContext> options)
            : base(options)
        {
        }

        public DbSet<Attribute> Attributes { get; set; }
        public DbSet<AttributeRelationship> AttributeRelationships { get; set; }

        protected override void OnModelCreating(ModelBuilder modelBuilder)
        {
            base.OnModelCreating(modelBuilder);
            modelBuilder.ApplyConfiguration(new AttributeRelationshipMap());

            modelBuilder.ApplyConfiguration(new AttributeMap());
            modelBuilder.Entity<Attribute>()
                .HasDiscriminator<int>("AttributeTypeId")
                .HasValue<StringAttribute>(7)
                ;            
        }
    }
```

SQL for creating tables and data that currently exists

```SQL
USE [database]
GO
/****** Object:  Table [dbo].[AttributeDisplayLabel]    Script Date: 30/09/2020 14:31:41 ******/
SET ANSI_NULLS ON
GO
SET QUOTED_IDENTIFIER ON
GO
CREATE TABLE [dbo].[AttributeDisplayLabel](
	[AttributeId] [int] NOT NULL,
	[CultureCode] [varchar](10) NOT NULL,
	[DisplayLabel] [nvarchar](250) NOT NULL,
 CONSTRAINT [PK_AttributeDisplayLabel] PRIMARY KEY CLUSTERED 
(
	[AttributeId] ASC,
	[CultureCode] ASC
)WITH (PAD_INDEX = OFF, STATISTICS_NORECOMPUTE = OFF, IGNORE_DUP_KEY = OFF, ALLOW_ROW_LOCKS = ON, ALLOW_PAGE_LOCKS = ON) ON [PRIMARY]
) ON [PRIMARY]
GO
/****** Object:  Table [dbo].[AttributeOptions]    Script Date: 30/09/2020 14:31:41 ******/
SET ANSI_NULLS ON
GO
SET QUOTED_IDENTIFIER ON
GO
CREATE TABLE [dbo].[AttributeOptions](
	[AttributeOptionId] [int] IDENTITY(1,1) NOT NULL,
	[AttributeId] [int] NOT NULL,
	[OptionIndex] [int] NOT NULL,
	[CultureCode] [varchar](8) NOT NULL,
	[Value] [nvarchar](1000) NOT NULL,
 CONSTRAINT [PK_AttributeOptions] PRIMARY KEY CLUSTERED 
(
	[AttributeOptionId] ASC
)WITH (PAD_INDEX = OFF, STATISTICS_NORECOMPUTE = OFF, IGNORE_DUP_KEY = OFF, ALLOW_ROW_LOCKS = ON, ALLOW_PAGE_LOCKS = ON) ON [PRIMARY]
) ON [PRIMARY]
GO
/****** Object:  Table [dbo].[AttributeRelationships]    Script Date: 30/09/2020 14:31:41 ******/
SET ANSI_NULLS ON
GO
SET QUOTED_IDENTIFIER ON
GO
CREATE TABLE [dbo].[AttributeRelationships](
	[AttributeRelationshipId] [int] IDENTITY(10000,1) NOT NULL,
	[ParentAttributeRelationshipId] [int] NULL,
	[AncestorAttributeId] [int] NOT NULL,
	[DescendantAttributeId] [int] NOT NULL,
	[AttributeRelationshipTypeId] [int] NOT NULL,
 CONSTRAINT [PK_AttributeRelationships] PRIMARY KEY CLUSTERED 
(
	[AttributeRelationshipId] ASC
)WITH (PAD_INDEX = OFF, STATISTICS_NORECOMPUTE = OFF, IGNORE_DUP_KEY = OFF, ALLOW_ROW_LOCKS = ON, ALLOW_PAGE_LOCKS = ON) ON [PRIMARY]
) ON [PRIMARY]
GO
/****** Object:  Table [dbo].[AttributeRelationshipTypes]    Script Date: 30/09/2020 14:31:41 ******/
SET ANSI_NULLS ON
GO
SET QUOTED_IDENTIFIER ON
GO
CREATE TABLE [dbo].[AttributeRelationshipTypes](
	[AttributeRelationshipTypeId] [int] NOT NULL,
	[Description] [varchar](100) NOT NULL,
 CONSTRAINT [PK_PropertyRelationshipTypes] PRIMARY KEY CLUSTERED 
(
	[AttributeRelationshipTypeId] ASC
)WITH (PAD_INDEX = OFF, STATISTICS_NORECOMPUTE = OFF, IGNORE_DUP_KEY = OFF, ALLOW_ROW_LOCKS = ON, ALLOW_PAGE_LOCKS = ON) ON [PRIMARY]
) ON [PRIMARY]
GO
/****** Object:  Table [dbo].[Attributes]    Script Date: 30/09/2020 14:31:41 ******/
SET ANSI_NULLS ON
GO
SET QUOTED_IDENTIFIER ON
GO
CREATE TABLE [dbo].[Attributes](
	[AttributeId] [int] IDENTITY(10000,1) NOT NULL,
	[AttributeTypeId] [int] NOT NULL,
	[PropertyName] [varchar](100) NOT NULL,
 CONSTRAINT [PK_Attributes] PRIMARY KEY CLUSTERED 
(
	[AttributeId] ASC
)WITH (PAD_INDEX = OFF, STATISTICS_NORECOMPUTE = OFF, IGNORE_DUP_KEY = OFF, ALLOW_ROW_LOCKS = ON, ALLOW_PAGE_LOCKS = ON) ON [PRIMARY]
) ON [PRIMARY]
GO
/****** Object:  Table [dbo].[AttributeTypes]    Script Date: 30/09/2020 14:31:41 ******/
SET ANSI_NULLS ON
GO
SET QUOTED_IDENTIFIER ON
GO
CREATE TABLE [dbo].[AttributeTypes](
	[AttributeTypeId] [int] NOT NULL,
	[Description] [varchar](100) NOT NULL,
 CONSTRAINT [PK_PropertyTypes] PRIMARY KEY CLUSTERED 
(
	[AttributeTypeId] ASC
)WITH (PAD_INDEX = OFF, STATISTICS_NORECOMPUTE = OFF, IGNORE_DUP_KEY = OFF, ALLOW_ROW_LOCKS = ON, ALLOW_PAGE_LOCKS = ON) ON [PRIMARY]
) ON [PRIMARY]
GO
INSERT [dbo].[AttributeDisplayLabel] ([AttributeId], [CultureCode], [DisplayLabel]) VALUES (10001, N'en-GB', N'Thickness')
GO
INSERT [dbo].[AttributeDisplayLabel] ([AttributeId], [CultureCode], [DisplayLabel]) VALUES (10001, N'fr-FR', N'Épaisseur')
GO
GO
SET IDENTITY_INSERT [dbo].[Attributes] ON 
GO
INSERT [dbo].[Attributes] ([AttributeId], [AttributeTypeId], [PropertyName]) VALUES (10001, 7, N'thickness')
GO
SET IDENTITY_INSERT [dbo].[Attributes] OFF
GO
INSERT [dbo].[AttributeTypes] ([AttributeTypeId], [Description]) VALUES (7, N'text')
GO

/****** Object:  Index [UQ_AttributeRelationships]    Script Date: 30/09/2020 14:31:41 ******/
ALTER TABLE [dbo].[AttributeRelationships] ADD  CONSTRAINT [UQ_AttributeRelationships] UNIQUE NONCLUSTERED 
(
	[ParentAttributeRelationshipId] ASC,
	[AncestorAttributeId] ASC,
	[DescendantAttributeId] ASC
)WITH (PAD_INDEX = OFF, STATISTICS_NORECOMPUTE = OFF, SORT_IN_TEMPDB = OFF, IGNORE_DUP_KEY = OFF, ONLINE = OFF, ALLOW_ROW_LOCKS = ON, ALLOW_PAGE_LOCKS = ON) ON [PRIMARY]
GO
ALTER TABLE [dbo].[AttributeDisplayLabel]  WITH CHECK ADD  CONSTRAINT [FK_AttributeDisplayLabel_Attributes] FOREIGN KEY([AttributeId])
REFERENCES [dbo].[Attributes] ([AttributeId])
GO
ALTER TABLE [dbo].[AttributeDisplayLabel] CHECK CONSTRAINT [FK_AttributeDisplayLabel_Attributes]
GO
ALTER TABLE [dbo].[AttributeOptions]  WITH NOCHECK ADD  CONSTRAINT [FK_AttributeOptions_Attributes] FOREIGN KEY([AttributeId])
REFERENCES [dbo].[Attributes] ([AttributeId])
GO
ALTER TABLE [dbo].[AttributeOptions] CHECK CONSTRAINT [FK_AttributeOptions_Attributes]
GO
ALTER TABLE [dbo].[AttributeRelationships]  WITH NOCHECK ADD  CONSTRAINT [FK_AncestorAttributeRelationships_Attributes] FOREIGN KEY([AncestorAttributeId])
REFERENCES [dbo].[Attributes] ([AttributeId])
GO
ALTER TABLE [dbo].[AttributeRelationships] CHECK CONSTRAINT [FK_AncestorAttributeRelationships_Attributes]
GO
ALTER TABLE [dbo].[AttributeRelationships]  WITH NOCHECK ADD  CONSTRAINT [FK_AttributeRelationships_AttributeRelationshipTypes] FOREIGN KEY([AttributeRelationshipTypeId])
REFERENCES [dbo].[AttributeRelationshipTypes] ([AttributeRelationshipTypeId])
GO
ALTER TABLE [dbo].[AttributeRelationships] CHECK CONSTRAINT [FK_AttributeRelationships_AttributeRelationshipTypes]
GO
ALTER TABLE [dbo].[AttributeRelationships]  WITH NOCHECK ADD  CONSTRAINT [FK_DescendantAttributeRelationships_Attributes] FOREIGN KEY([DescendantAttributeId])
REFERENCES [dbo].[Attributes] ([AttributeId])
GO
ALTER TABLE [dbo].[AttributeRelationships] CHECK CONSTRAINT [FK_DescendantAttributeRelationships_Attributes]
GO
ALTER TABLE [dbo].[AttributeRelationships]  WITH NOCHECK ADD  CONSTRAINT [FK_ParentAttributeRelationships_AttributeRelationships] FOREIGN KEY([ParentAttributeRelationshipId])
REFERENCES [dbo].[AttributeRelationships] ([AttributeRelationshipId])
GO
ALTER TABLE [dbo].[AttributeRelationships] CHECK CONSTRAINT [FK_ParentAttributeRelationships_AttributeRelationships]
GO
ALTER TABLE [dbo].[Attributes]  WITH NOCHECK ADD  CONSTRAINT [FK_Attributes_AttributeTypes] FOREIGN KEY([AttributeTypeId])
REFERENCES [dbo].[AttributeTypes] ([AttributeTypeId])
GO
ALTER TABLE [dbo].[Attributes] CHECK CONSTRAINT [FK_Attributes_AttributeTypes]
GO

```

### Stack traces

```
Microsoft.Data.SqlClient.SqlException
  HResult=0x80131904
  Message=Invalid column name 'AttributeId'.
  Source=Core Microsoft SqlClient Data Provider
  StackTrace:
   at Microsoft.Data.SqlClient.SqlConnection.OnError(SqlException exception, Boolean breakConnection, Action`1 wrapCloseInAction)
   at Microsoft.Data.SqlClient.SqlInternalConnection.OnError(SqlException exception, Boolean breakConnection, Action`1 wrapCloseInAction)
   at Microsoft.Data.SqlClient.TdsParser.ThrowExceptionAndWarning(TdsParserStateObject stateObj, Boolean callerHasConnectionLock, Boolean asyncClose)
   at Microsoft.Data.SqlClient.TdsParser.TryRun(RunBehavior runBehavior, SqlCommand cmdHandler, SqlDataReader dataStream, BulkCopySimpleResultSet bulkCopyHandler, TdsParserStateObject stateObj, Boolean& dataReady)
   at Microsoft.Data.SqlClient.SqlDataReader.TryConsumeMetaData()
   at Microsoft.Data.SqlClient.SqlDataReader.get_MetaData()
   at Microsoft.Data.SqlClient.SqlCommand.FinishExecuteReader(SqlDataReader ds, RunBehavior runBehavior, String resetOptionsString, Boolean isInternal, Boolean forDescribeParameterEncryption, Boolean shouldCacheForAlwaysEncrypted)
   at Microsoft.Data.SqlClient.SqlCommand.RunExecuteReaderTds(CommandBehavior cmdBehavior, RunBehavior runBehavior, Boolean returnStream, Boolean isAsync, Int32 timeout, Task& task, Boolean asyncWrite, Boolean inRetry, SqlDataReader ds, Boolean describeParameterEncryptionRequest)
   at Microsoft.Data.SqlClient.SqlCommand.RunExecuteReader(CommandBehavior cmdBehavior, RunBehavior runBehavior, Boolean returnStream, TaskCompletionSource`1 completion, Int32 timeout, Task& task, Boolean& usedCache, Boolean asyncWrite, Boolean inRetry, String method)
   at Microsoft.Data.SqlClient.SqlCommand.RunExecuteReader(CommandBehavior cmdBehavior, RunBehavior runBehavior, Boolean returnStream, String method)
   at Microsoft.Data.SqlClient.SqlCommand.ExecuteReader(CommandBehavior behavior)
   at Microsoft.Data.SqlClient.SqlCommand.ExecuteDbDataReader(CommandBehavior behavior)
   at System.Data.Common.DbCommand.ExecuteReader()
   at Microsoft.EntityFrameworkCore.Storage.RelationalCommand.ExecuteReader(RelationalCommandParameterObject parameterObject)
   at Microsoft.EntityFrameworkCore.Query.Internal.SingleQueryingEnumerable`1.Enumerator.InitializeReader(DbContext _, Boolean result)
   at Microsoft.EntityFrameworkCore.SqlServer.Storage.Internal.SqlServerExecutionStrategy.Execute[TState,TResult](TState state, Func`3 operation, Func`3 verifySucceeded)
   at Microsoft.EntityFrameworkCore.Query.Internal.SingleQueryingEnumerable`1.Enumerator.MoveNext()
   at System.Collections.Generic.List`1..ctor(IEnumerable`1 collection)
   at System.Linq.Enumerable.ToList[TSource](IEnumerable`1 source)
   at VictorianPlumbing.ProductInformationService.Application.Attributes.AttributeCreator.Create(CreateAttributeRequestDto request) in C:\Code\ProductInformation\ProductInformationService\Application\Attributes\AttributeCreator.cs:line 60
   at VictorianPlumbing.ProductInformation.WebApi.Controllers.AttributesController.Create(CreateAttributeRequestDto request) in C:\Code\ProductInformation\ProductInformationService\WebApi\Controllers\AttributesController.cs:line 27
   at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.SyncActionResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<<InvokeActionMethodAsync>g__Logged|12_1>d.MoveNext()

  This exception was originally thrown at this call stack:
    [External Code]
    VictorianPlumbing.ProductInformationService.Application.Attributes.AttributeCreator.Create(VictorianPlumbing.ProductInformationService.ClientDtos.CreateAttributeRequestDto) in AttributeCreator.cs
    VictorianPlumbing.ProductInformation.WebApi.Controllers.AttributesController.Create(VictorianPlumbing.ProductInformationService.ClientDtos.CreateAttributeRequestDto) in AttributesController.cs
    [External Code]
```

Equivalent Mapping code in EF 6.3 - domain code and context are basically the same
```C#
public class AttributeMap : EntityTypeConfiguration<Attribute>
    {
        public AttributeMap()
            : this("dbo")
        {
        }

        public AttributeMap(string schema)
        {
            ToTable("Attributes", schema);
            HasKey(x => x.AttributeId);

            Property(x => x.AttributeId).HasColumnName(@"AttributeId").HasColumnType("int").IsRequired().HasDatabaseGeneratedOption(DatabaseGeneratedOption.Identity);
            Property(x => x.LabelId).HasColumnName(@"LabelId").HasColumnType("int").IsRequired();

            // Foreign keys
            HasRequired(a => a.Label).WithMany(b => b.Attributes).HasForeignKey(c => c.LabelId).WillCascadeOnDelete(false);

            Ignore(x => x.StructuralDescendants);
        }
    }

public class AttributeRelationshipMap : EntityTypeConfiguration<AttributeRelationship>
    {
        public AttributeRelationshipMap()
            : this("dbo")
        {
        }

        public AttributeRelationshipMap(string schema)
        {
            ToTable("AttributeRelationships", schema);
            HasKey(x => x.AttributeRelationshipId);

            Property(x => x.AttributeRelationshipId).HasColumnName(@"AttributeRelationshipId").HasColumnType("int").IsRequired().HasDatabaseGeneratedOption(DatabaseGeneratedOption.Identity);
            Property(x => x.ParentAttributeRelationshipId).HasColumnName(@"ParentAttributeRelationshipId").HasColumnType("int").IsOptional();
            Property(x => x.AncestorAttributeId).HasColumnName(@"AncestorAttributeId").HasColumnType("int").IsRequired().HasDatabaseGeneratedOption(DatabaseGeneratedOption.None);
            Property(x => x.DescendantAttributeId).HasColumnName(@"DescendantAttributeId").HasColumnType("int").IsRequired().HasDatabaseGeneratedOption(DatabaseGeneratedOption.None);
            Property(x => x.AttributeRelationshipTypeId).HasColumnName(@"AttributeRelationshipTypeId").HasColumnType("int").IsRequired();

            // Foreign keys
            HasOptional(a => a.ParentAttributeRelationship).WithMany(b => b.ChildAttributeRelationships).HasForeignKey(c => c.ParentAttributeRelationshipId).WillCascadeOnDelete(false);
            HasRequired(a => a.Attribute).WithMany(b => b.AncestorOfRelationships).HasForeignKey(c => c.AncestorAttributeId).WillCascadeOnDelete(false);
            HasRequired(a => a.DescendantAttribute).WithMany(b => b.DescendantOfRelationships).HasForeignKey(c => c.DescendantAttributeId).WillCascadeOnDelete(false);
            HasRequired(a => a.AttributeRelationshipType).WithMany(b => b.AttributeRelationships).HasForeignKey(c => c.AttributeRelationshipTypeId).WillCascadeOnDelete(false);
        }
    }
```

### Provider and version information

EF Core version: 5.0 rc1
Database provider: Microsoft.EntityFrameworkCore.SqlServer
Target framework: .NET 5.0
Operating system: Windows 19
IDE: Visual Studio 16.8.0 Preview 3.2

**Comment by smitpatel** [comment]
It is not query issue, model building is adding additional column. Query just happens to be a place where it throws.

Inspecting code snippets above,
For `AttributeRelationship`, in EF Core you are mapping navigation `AncestorAttribute` in a relationship where FK Id is `AncestorAttributeId`. Same FK property is being used in EF6 configuration with navigation `Attribute` but that does not exist on the class. Can you share a runnable repro code?

**Comment by davemanton** [comment]
Thanks for your time and also sorry to waste it.

To answer your query I took the ef 6 code from a different branch where I had named the differently however they were mapped correctly and basically the same.

I found the issue though.  I had a property on the attribute (which I hadn't submitted as part of snippets) which was an IEnumerable<AttributeRelationship> that I use for navigating relationships in the attribute object.  I hadn't added an ignore on the mapping for the attribute.  When I did the query started to work correctly.  It would seem that I need to be much more careful about ignoring properties in EF Core than I do in EF 6.3
